spring-xd-test/src/main/java/org/springframework/xd/test/kafka/KafkaTestSupport.java

```diff
@@ -49,81 +62,99 @@
 
 	private Properties brokerConfig = TestUtils.createBrokerConfig(0, TestUtils.choosePort());
 
-	public KafkaTestSupport() {
-		super("KAFKA");
+	static {
+		embedded = "true".equals(System.getProperty(XD_KAFKA_TEST_EMBEDDED));
+		log.info(String.format("Testing with %s Kafka broker", embedded ? "embedded" : "external"));
 	}
 
-	public void setBrokerConfig(Properties brokerConfig) {
-		this.brokerConfig = brokerConfig;
+	public KafkaTestSupport() {
+		super("KAFKA");
 	}
 
 	public String getZkConnectString() {
-		return zookeeper.getConnectString();
+		if (embedded) {
+			return zookeeper.getConnectString();
+		}
+		else {
+			return DEFAULT_ZOOKEEPER_CONNECT;
+		}
 	}
 
 	public ZkClient getZkClient() {
 		return this.zkClient;
 	}
 
-	public EmbeddedZookeeper getZookeeper() {
-		return zookeeper;
-	}
-
-	public KafkaServer getKafkaServer() {
-		return kafkaServer;
-	}
-
 	public String getBrokerAddress() {
-		return getKafkaServer().config().hostName() + ":" + getKafkaServer().config().port();
+		if (embedded) {
+			return kafkaServer.config().hostName() + ":" + kafkaServer.config().port();
+		}
+		else {
+			return DEFAULT_KAFKA_CONNECT;
+		}
 	}
 
 	@Override
 	protected void obtainResource() throws Exception {
-		try {
-			log.debug("Starting Zookeeper");
-			zookeeper = new EmbeddedZookeeper(TestZKUtils.zookeeperConnect());
-			log.debug("Started Zookeeper at " + zookeeper.getConnectString());
+		if (embedded) {
+			try {
+				log.debug("Starting Zookeeper");
+				zookeeper = new EmbeddedZookeeper(TestZKUtils.zookeeperConnect());
+				log.debug("Started Zookeeper at " + zookeeper.getConnectString());
+			}
+			catch (Exception e) {
+				throw new RuntimeException("Issues creating the ZK server", e);
+			}
+			try {
+				int zkConnectionTimeout = 6000;
+				int zkSessionTimeout = 6000;
+				zkClient = new ZkClient(getZkConnectString(), zkSessionTimeout, zkConnectionTimeout, ZKStringSerializer$.MODULE$);
+			}
+			catch (Exception e) {
+				zookeeper.shutdown();
+				throw new RuntimeException("Issues creating the ZK client", e);
+			}
+			try {
+				log.debug("Creating Kafka server");
+				Properties brokerConfigProperties = brokerConfig;
+				kafkaServer = TestUtils.createServer(new KafkaConfig(brokerConfigProperties), SystemTime$.MODULE$);
+				log.debug("Created Kafka server at " + kafkaServer.config().hostName() + ":" + kafkaServer.config().port());
+			}
+			catch (Exception e) {
+				zookeeper.shutdown();
+				zkClient.close();
+				throw new RuntimeException("Issues creating the Kafka server", e);
+			}
```

Comment:
Nit picking here, but why wrap all of these Exceptions in RuntimeException if the method is throwing Exception anyways?